Src/LogData.c

```diff
@@ -36,316 +73,209 @@ void writeToEEPROM(uint8_t* buffer, uint16_t bufferSize, uint16_t memAddress, ui
 {
 	while(HAL_I2C_GetState(&hi2c1) != HAL_I2C_STATE_READY){};
 	while (HAL_I2C_IsDeviceReady(&hi2c1, deviceAddress<<1, 3, timeout) != HAL_OK){};
 }
 
-void writeAllDataToEEPROM(AllData* data, uint16_t memAddress, uint16_t timeout)
+/**
+ * @brief puts a logEntry into a char array and sends to EEPROM for writing
+ * @param memAddress, the address at which the EEPROM is instructed to start writing
+ */
+void writeLogEntryToEEPROM(uint16_t memAddress)
 {
-    char dataToWrite[200]; 
-
-    int index = 0;
-    dataToWrite[0] = data->accelGyroMagnetismData_ ->accelX_;
-    dataToWrite[4] = data->accelGyroMagnetismData_ ->accelY_;
-    dataToWrite[8] = data->accelGyroMagnetismData_ ->accelZ_;
-    dataToWrite[12] = data->accelGyroMagnetismData_ ->gyroX_;
-    dataToWrite[16] = data->accelGyroMagnetismData_ ->gyroY_;
-    dataToWrite[20] = data->accelGyroMagnetismData_ ->gyroZ_;
-    dataToWrite[24] = data->accelGyroMagnetismData_ ->magnetoX_;
-    dataToWrite[28] = data->accelGyroMagnetismData_ ->magnetoY_;
-    dataToWrite[32] = data->accelGyroMagnetismData_ ->magnetoZ_;
-
-    dataToWrite[36] = data->barometerData_ ->pressure_;
-    dataToWrite[40] = data->barometerData_ ->temperature_;
-
-    dataToWrite[44] = data->combustionChamberPressureData_ ->pressure_;
-
-    dataToWrite[48] = data->gpsData_ ->time_;
-    dataToWrite[52] = data->gpsData_->latitude_.degrees_;
-    dataToWrite[56] = data->gpsData_->latitude_.minutes_;
-    dataToWrite[60] = data->gpsData_->longitude_.degrees_;
-    dataToWrite[64] = data->gpsData_->longitude_.minutes_;
-
-    dataToWrite[68] = data->gpsData_->antennaAltitude_.altitude_;
-    dataToWrite[72] = data->gpsData_->geoidAltitude_.altitude_;
-    dataToWrite[76] = data->gpsData_->totalAltitude_.altitude_;
-
-
-    dataToWrite[80] = data->oxidizerTankPressureData_ ->pressure_;
-
-    checkEEPROMBlocking(uint16_t timeout);
-    writeToEEPROM(dataToWrite, 80, uint16_t memAddress, uint16_t timeout);
-    // need: memAddress and timeout??????
-    // increment address
+    char dataToWrite[logEntrySize]; 
+
+    dataToWrite[0] = theLogEntry.accelX;
+    dataToWrite[4] = theLogEntry.accelY;
+    dataToWrite[8] = theLogEntry.accelZ;
+    dataToWrite[12] = theLogEntry.gyroX;
+    dataToWrite[16] = theLogEntry.gyroY;
+    dataToWrite[20] = theLogEntry.gyroZ;
+    dataToWrite[24] = theLogEntry.magnetoX;
+    dataToWrite[28] = theLogEntry.magnetoY;
+    dataToWrite[32] = theLogEntry.magnetoZ;
+    dataToWrite[36] = theLogEntry.barometerPressure;
+    dataToWrite[40] = theLogEntry.barometerTemperature;
+    dataToWrite[44] = theLogEntry.combustionChamberPressure;
+    dataToWrite[48] = theLogEntry.oxidizerTankPressure;
+    dataToWrite[52] = theLogEntry.gps_time;
+    dataToWrite[56] = theLogEntry.latitude_degrees;
+    dataToWrite[60] = theLogEntry.latitude_minutes;
+    dataToWrite[64] = theLogEntry.longitude_degrees;
+    dataToWrite[68] = theLogEntry.longitude_minutes;
+    dataToWrite[72] = theLogEntry.altitude;
```

Comment:
Missing `antennaAltitude` and `geoidAltitude`. Is there a reason we wouldn't want to log these?

Reply:
See next comment